### ISSUE TITLE
Translating configurations from a model in OpenNMT-Py to OpenNMT-tf

### ISSUE THREAD

**Issue by sajeeth1009** [OpenNMT/OpenNMT-tf#417]
HI,

This is more of a question than an issue. I was using OpenNMT-Py to train my translation model using the following command
`python train.py -data data/nmt_model -save_model ./nmt_model -epoch 70 -layers 4 -batch_size 64 -word_vec_size 1024 -learning_rate 0.001 -learning_rate_decay 1 -start_decay_at -1 -rnn_size 1024 -optim adam -input_feed 3 -dropout 0.3
`

This works on tokenized inputs at a character level looking something like this
`l i b r o r u m <SPACE> m u l t i t u`

The resulting Model Parameters in NMT-Py is as follows:
`
NMTModel(
  (encoder): RNNEncoder(
    (embeddings): Embeddings(
      (make_embedding): Sequential(
        (emb_luts): Elementwise(
          (0): Embedding(75, 1024, padding_idx=1)
        )
      )
    )
    (rnn): LSTM(1024, 1024, num_layers=4, dropout=0.3)
  )
  (decoder): InputFeedRNNDecoder(
    (embeddings): Embeddings(
      (make_embedding): Sequential(
        (emb_luts): Elementwise(
          (0): Embedding(98, 1024, padding_idx=1)
        )
      )
    )
    (dropout): Dropout(p=0.3)
    (rnn): StackedLSTM(
      (dropout): Dropout(p=0.3)
      (layers): ModuleList(
        (0): LSTMCell(2048, 1024)
        (1): LSTMCell(1024, 1024)
        (2): LSTMCell(1024, 1024)
        (3): LSTMCell(1024, 1024)
      )
    )
    (attn): GlobalAttention(
      (linear_in): Linear(in_features=1024, out_features=1024, bias=False)
      (linear_out): Linear(in_features=2048, out_features=1024, bias=False)
      (sm): Softmax()
      (tanh): Tanh()
    )
  )
  (generator): Sequential(
    (0): Linear(in_features=1024, out_features=98, bias=True)
    (1): LogSoftmax()
  )
)`

I would like to train the exact same configurations via OpenNMT-Tf. And this is what i have done so far:

**data.yml**

`
model_dir: run/
data:
  train_features_file: src-train.txt
  train_labels_file: tgt-train.txt
  eval_features_file: src-val.txt
  eval_labels_file: tgt-val.txt
  source_words_vocabulary: src-vocab.txt
  target_words_vocabulary: tgt-vocab.txt

params:
  optimizer: AdamOptimizer  
  learning_rate: 0.001
  start_decay_steps: -1
  decay_params:
    decay_rate: 1

train:
  batch_size: 64
  save_checkpoints_steps: 1000
  keep_checkpoint_max: 50
  save_summary_steps: 100
  single_pass: true

eval:
  batch_size: 64
  eval_delay: 7200
  save_eval_predictions: true
  external_evaluators: BLEU
  exporters: last
`

**Model** 
`
import tensorflow as tf
import opennmt as onmt

def model():
  return onmt.models.SequenceToSequence(
      source_inputter=onmt.inputters.WordEmbedder(
          vocabulary_file_key="source_words_vocabulary",
          embedding_size=1024,
          dtype=tf.float32),
      target_inputter=onmt.inputters.WordEmbedder(
          vocabulary_file_key="target_words_vocabulary",
          embedding_size=1024,
          dtype=tf.float32),
      encoder=onmt.encoders.UnidirectionalRNNEncoder(
          num_layers=4,
          num_units=1024,
          cell_class=tf.contrib.rnn.LSTMCell,
          dropout=0.3,
          residual_connections=False),
      decoder=onmt.decoders.AttentionalRNNDecoder(
          num_layers=4,
          num_units=1024,
          bridge=onmt.layers.CopyBridge(),
          attention_mechanism_class=tf.contrib.seq2seq.LuongAttention,
          cell_class=tf.contrib.rnn.LSTMCell,
          dropout=0.3,
          residual_connections=False))`

After passing the dataset for n times to mimic n epochs, i still get far inferior results that the original OpenNMT-Py implementation. Perhaps im missing something or making a really stupid mistake somewhere. Are my configurations correct? Any help would be greatly appreciated

**Comment by guillaumekln** [comment]
Hi,

The configuration looks about right.

* How does the results compare?
* Can you share the command line you used for inference with OpenNMT-py? OpenNMT-py is doing beam search by default while OpenNMT-tf does not.

**Comment by guillaumekln** [comment]
Closing due to lack of activity. Feel free to reopen it if you feel the issue has not been resolved.
